list_test.go

```diff
@@ -58,13 +70,82 @@ func TestList(t *testing.T) {
 	testDecodeFramework(t, "customReplyTypedVariableListHasNull", []interface{}{new(A0), new(A1), nil})
 	testDecodeFramework(t, "customReplyUntypedFixedListHasNull", []interface{}{new(A0), new(A1), nil})
 	testDecodeFramework(t, "customReplyUntypedVariableListHasNull", []interface{}{new(A0), new(A1), nil})
+
+	testDecodeFramework(t, "customReplyTypedFixedList_A0", []*A0{new(A0), new(A0), nil})
+	testDecodeFramework(t, "customReplyTypedVariableList_A0", []*A0{new(A0), new(A0), nil})
+
+	testDecodeFramework(t, "customReplyTypedFixedList_int", []int32{1, 2, 3})
+	testDecodeFramework(t, "customReplyTypedVariableList_int", []int32{1, 2, 3})
+
+	testDecodeFramework(t, "customReplyTypedFixedList_long", []int64{1, 2, 3})
+	testDecodeFramework(t, "customReplyTypedVariableList_long", []int64{1, 2, 3})
+
+	testDecodeFramework(t, "customReplyTypedFixedList_float", []float64{1, 2, 3})
+	testDecodeFramework(t, "customReplyTypedVariableList_float", []float64{1, 2, 3})
+
+	testDecodeFramework(t, "customReplyTypedFixedList_double", []float64{1, 2, 3})
+	testDecodeFramework(t, "customReplyTypedVariableList_double", []float64{1, 2, 3})
+
+	testDecodeFramework(t, "customReplyTypedFixedList_short", []int32{1, 2, 3})
+	testDecodeFramework(t, "customReplyTypedVariableList_short", []int32{1, 2, 3})
+
+	testDecodeFramework(t, "customReplyTypedFixedList_char", []string{"1", "2", "3"})
+	testDecodeFramework(t, "customReplyTypedVariableList_char", []string{"1", "2", "3"})
+
+	testDecodeFramework(t, "customReplyTypedFixedList_boolean", []bool{true, false, true})
+	testDecodeFramework(t, "customReplyTypedVariableList_boolean", []bool{true, false, true})
+
+	testDecodeFramework(t, "customReplyTypedFixedList_date", []time.Time{time.Unix(1560864, 0),
+		time.Unix(1560864, 0), time.Unix(1560864, 0)})
+	testDecodeFramework(t, "customReplyTypedVariableList_date", []time.Time{time.Unix(1560864, 0),
+		time.Unix(1560864, 0), time.Unix(1560864, 0)})
 }
 
 func TestListEncode(t *testing.T) {
+	RegisterPOJOs(new(A0))
+
 	testJavaDecode(t, "argUntypedFixedList_0", []interface{}{})
 	testJavaDecode(t, "argUntypedFixedList_1", []interface{}{"1"})
 	testJavaDecode(t, "argUntypedFixedList_7", []interface{}{"1", "2", "3", "4", "5", "6", "7"})
 	testJavaDecode(t, "argUntypedFixedList_8", []interface{}{"1", "2", "3", "4", "5", "6", "7", "8"})
 
 	testJavaDecode(t, "customArgUntypedFixedListHasNull", []interface{}{new(A0), new(A1), nil})
+
+	testJavaDecode(t, "customArgTypedFixedList", []*A0{new(A0)})
+
+	testJavaDecode(t, "argTypedFixedList_0", []string{})
+	testJavaDecode(t, "argTypedFixedList_7", []string{"1", "2", "3", "4", "5", "6", "7"})
```

Comment:
Add `argTypedFixedList_1` and `argTypedFixedList_8` may be better.

Reply:
What does 1 and 8 prove? I think 0 and 7 already represent all case.

Reply:
@fangyincheng Yes, I agree with you. They are the same.